File: framework/java/src/main/java/soda/unittest/job/OutputData.java
package soda.unittest.job;

public class OutputData {

	public int id;
	
	public boolean success;
	
	public Object result;
	
	public double elapse;
	
	public void setResult(Object res) {
		this.result = res;
	}
	
	public <T> T getResult(Class<T> klass) {
		return klass.cast(result);
	}
	
}

File: framework/scala/src/main/scala/soda/scala/unittest/OutputData.scala
package soda.scala.unittest

class OutputData {

    var id: Int = 0

    var success: Boolean = false

    var result: Object = null

    var elapse: Double = 0.0

    def setResult(res: Object): Unit = result = res

    def getResult[T](klass: Class[T]): T = klass.cast(result)

}
